mapper/thin_edge_json/src/json.rs

```diff
@@ -289,6 +298,101 @@ mod tests {
             .and_hms(0, 0, 0)
     }
 
+    #[test]
+    fn valid_short_time_stamp() {
+        let input = r#"{
+            "time" : "2021-04-30T17:03:14.123+02:00"
+        }"#;
+        let json = json::parse(&input).unwrap();
+
+        let expected_output = FixedOffset::east(2 * 3600)
+            .ymd(2021, 04, 30)
+            .and_hms_milli(17, 03, 14, 123);
+
+        let time = json.entries().nth(0).unwrap();
+        let output = ThinEdgeJson::check_timestamp_for_iso8601_complaint(time.1).unwrap();
+        assert_eq!(output, expected_output);
+    }
+
+    #[test]
+    fn valid_string_time_stamp() {
+        let input = r#"{
+            "time" : "2021-04-30T17:03:14.123456789+02:00"
+        }"#;
+
+        let json = json::parse(&input).unwrap();
+        let expected_output = FixedOffset::east(2 * 3600)
+            .ymd(2021, 04, 30)
+            .and_hms_nano(17, 03, 14, 123456789);
+
+        let time = json.entries().nth(0).unwrap();
+        let output = ThinEdgeJson::check_timestamp_for_iso8601_complaint(time.1).unwrap();
+        assert_eq!(output, expected_output);
+    }
+
+    #[test]
+    fn invalid_time_stamp() {
+        let input = r#"{
+            "time" : "2013-06-2217:03:14.000658767+02:00"
+        }"#;
+        let expected_error = r#"Invalid ISO8601 timestamp (expected YYYY-MM-DDThh:mm:ss.sss.±hh:mm): "2013-06-2217:03:14.000658767+02:00": input contains invalid characters"#;
+        let json = json::parse(&input).unwrap();
+        let time = json.entries().nth(0).unwrap();
+        let output = ThinEdgeJson::check_timestamp_for_iso8601_complaint(time.1).unwrap_err();
+        assert_eq!(output.to_string(), expected_error);
+    }
+
+    #[test]
+    fn check_from_json_str_short_timestamp() {
+        let input = r#"{
+            "temperature" : 25
+        }"#;
+        let timestamp = FixedOffset::east(2 * 3600)
+            .ymd(2021, 04, 30)
+            .and_hms_milli(17, 03, 14, 123);
+
+        //prepare expected output
+        let tvalues: ThinEdgeValue = ThinEdgeValue::Single(SingleValueMeasurement {
+            name: "temperature".into(),
+            value: 25.0,
+        });
+        let tjson: ThinEdgeJson = ThinEdgeJson {
+            timestamp,
+            values: vec![tvalues],
+        };
+        let expected_string = format!("{:#?}", tjson);
+
+        let output = ThinEdgeJson::from_str(input, timestamp).unwrap();
+        let output_string = format!("{:#?}", output);
+
+        assert_eq!(output_string, expected_string);
+    }
+
+    #[test]
+    fn check_from_json_str_string_local_time_now() {
+        let input = r#"{
+            "temperature" : 25
+        }"#;
+        let local_time_now: DateTime<Local> = Local::now();
+        let timestamp = local_time_now.with_timezone(local_time_now.offset());

```

Comment:
Could you replace this call with the `thin_edge_json::measurement::current_timestamp()` method so that we always test that the output of that API is also always valid?

Reply:
done